src/main/java/com/unzer/payment/models/CardTransactionData.java

```diff
@@ -0,0 +1,24 @@
+package com.unzer.payment.models;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.unzer.payment.enums.RecurrenceType;
+
+@JsonTypeName("card")
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+public class CardTransactionData {
+
+    public CardTransactionData(RecurrenceType recurrenceType) {
+        this.recurrenceType = recurrenceType;
+    }
+
+    RecurrenceType recurrenceType;
```

Comment:
recurrenceType to be declared at the top of the class, before constructor

see https://app.codacy.com/gh/unzerdev/java-sdk/pullRequest?prid=7995522

Reply:
done